### PR TITLE
Support locking functions

### PR DESCRIPTION
**Pull request by systay**
Closes: #6367

### CODE REVIEW

File: go/vt/vttablet/tabletserver/planbuilder/plan_test.go [hunk 1]
```diff
@@ -72,7 +72,76 @@ func TestPlan(t *testing.T) {
...
+				} else {
+					bout, _ := json.MarshalIndent(plan, "", "  ")
+					out = string(bout)
+				}
+				fmt.Printf("\"%s\"\n%s\n\n", tcase.input, out)
+			}
+			// In Reserved Connection, plan will be built.
+			plan, err = Build(statement, testSchema, true /* isReservedConn */)
+			require.NoError(t, err)
+			require.NotEmpty(t, plan)
```
Comment (systay) [inline go/vt/vttablet/tabletserver/planbuilder/plan_test.go:128]: I would really like to see the plan in the txt file, not just checking that it's not empty.

Comment (harshit-gangal) [reply go/vt/vttablet/tabletserver/planbuilder/plan_test.go:128]: It is vttablet plan. I can add it. I believe VTGate side plan is of more interest.

File: go/vt/vttablet/tabletserver/query_engine_test.go [hunk 2]
```diff
@@ -98,7 +98,7 @@ func TestGetPlanPanicDuetoEmptyQuery(t *testing.T) {
 
 	ctx := context.Background()
 	logStats := tabletenv.NewLogStats(ctx, "GetPlanStats")
-	_, err := qe.GetPlan(ctx, logStats, "", false)
+	_, err := qe.GetPlan(ctx, logStats, "", false, false /* inReservedConn */)
```
Comment (deepthi) [inline go/vt/vttablet/tabletserver/query_engine_test.go:101]: does it make sense to add tests here with true?

Comment (harshit-gangal) [reply go/vt/vttablet/tabletserver/query_engine_test.go:101]: query_engine does not work with reserved connections.

**Pull request merged**